## WEATHER STATISTICS PROJECT
##### MEMBERS : ESPINOUX Jules, FALCOZ Nils, GRUNEWALD Augustin, PICQ Florian

#### 1 - INTRODUCTION TO DATA - NUMERICAL & GRAPHICAL SUMMARIES

In this introduction part we're going to study the dataset, the different categories, size of the set, ... 

In [35]:
data_weather <- read.table("data/donnees-synop-essentielles-omm.csv",
                        header=TRUE,
                        sep=";",
                        quote = "\"") # treat text inside "..." as a single field, even if it contains ;

In [45]:
head(data_weather)
dim(data_weather)
summary(data_weather)

ID.OMM.station Date                      Pression.au.niveau.mer
1  7181          2016-07-12T11:00:00+02:00 101410                
2 81405          2017-01-08T07:00:00+01:00 101140                
3  7627          2017-01-08T13:00:00+01:00 103380                
4 78894          2017-01-09T13:00:00+01:00 101760                
5  7020          2017-01-12T13:00:00+01:00 100580                
6  7280          2017-12-23T13:00:00+01:00 103830                
  Variation.de.pression.en.3.heures Type.de.tendance.barométrique
1  -30                               8                           
2 -140                               8                           
3  -40                               8                           
4    0                              NA                           
5 -710                               8                           
6  -10                               0                           
  Direction.du.vent.moyen.10.mn Vitesse.du.vent.moyen.10.mn Température
1 230                            2.9                        293.35     
2 240                            0.5                        297.25     
3  30                            2.1                        277.55     
4 100                            2.3                        298.15     
5 210                           11.9                        282.05     
6  10                            2.7                        280.75     
  Point.de.rosée Humidité ⋯ Altitude communes..name.         communes..code.
1 286.55         65       ⋯ 336      Thuilley-aux-Groseilles 54523          
2 296.25         94       ⋯   4      Matoury                 97307          
3 269.75         57       ⋯ 414      Lorp-Sentaraille        09289          
4 291.35         66       ⋯  44                                             
5 280.15         88       ⋯   6      La Hague                50041          
6 278.35         85       ⋯ 219      Ouges                   21473          
  EPCI..name.                              EPCI..code. department..name. 
1 CC du Pays de Colombey et du Sud Toulois 245400510   Meurthe-et-Moselle
2 CA du Centre Littoral                    249730045   Guyane            
3 CC Couserans-Pyrénées                    200067940   Ariège            
4                                                 NA                     
5 CA du Cotentin                           200067205   Manche            
6 Dijon Métropole                          242100410   Côte-d'Or         
  department..code. region..name.           region..code. mois_de_l_annee
1 54                Grand Est               44             7             
2 973               Guyane                   3             1             
3 09                Occitanie               76             1             
4                                           NA             1             
5 50                Normandie               28             1             
6 21                Bourgogne-Franche-Comté 27            12

[1] 1719567      82

 ID.OMM.station      Date           Pression.au.niveau.mer
 Min.   : 7005   Length:1719567     Min.   : 93850        
 1st Qu.: 7207   Class :character   1st Qu.:101180        
 Median : 7577   Mode  :character   Median :101600        
 Mean   :26062                      Mean   :101577        
 3rd Qu.:61976                      3rd Qu.:102090        
 Max.   :89642                      Max.   :107570        
                                    NA's   :94370         
 Variation.de.pression.en.3.heures Type.de.tendance.barométrique
 Min.   :-2840.00                  Min.   : 0.0                 
 1st Qu.:  -60.00                  1st Qu.: 2.0                 
 Median :   10.00                  Median : 3.0                 
 Mean   :   15.39                  Mean   : 4.3                 
 3rd Qu.:   90.00                  3rd Qu.: 7.0                 
 Max.   : 2990.00                  Max.   :10.0                 
 NA's   :40694                     NA's   :68236                
 Directi

- First thing to notice is the huge size of the dataset : dataframe with 82 variables and 1,719,567 data.
- We also have a numerical summary of all the numerical data (summary don't give information for string variables such as the names of the station, the region, ...)

In [43]:
# Showing the labels
labels(data_weather)[2]

[[1]]
 [1] "ID.OMM.station"                                           
 [2] "Date"                                                     
 [3] "Pression.au.niveau.mer"                                   
 [4] "Variation.de.pression.en.3.heures"                        
 [5] "Type.de.tendance.barométrique"                            
 [6] "Direction.du.vent.moyen.10.mn"                            
 [7] "Vitesse.du.vent.moyen.10.mn"                              
 [8] "Température"                                              
 [9] "Point.de.rosée"                                           
[10] "Humidité"                                                 
[11] "Visibilité.horizontale"                                   
[12] "Temps.présent"                                            
[13] "Temps.passé.1"                                            
[14] "Temps.passé.2"                                            
[15] "Nebulosité.totale"                                        
[16] "Nébulosité..des.nuages.de.l..étage.inférieur"             
[17] "Hauteur.de.la.base.des.nuages.de.l.étage.inférieur"       
[18] "Type.des.nuages.de.l.étage.inférieur"                     
[19] "Type.des.nuages.de.l.étage.moyen"                         
[20] "Type.des.nuages.de.l.étage.supérieur"                     
[21] "Pression.station"                                         
[22] "Niveau.barométrique"                                      
[23] "Géopotentiel"                                             
[24] "Variation.de.pression.en.24.heures"                       
[25] "Température.minimale.sur.12.heures"                       
[26] "Température.minimale.sur.24.heures"                       
[27] "Température.maximale.sur.12.heures"                       
[28] "Température.maximale.sur.24.heures"                       
[29] "Température.minimale.du.sol.sur.12.heures"                
[30] "Méthode.de.mesure.Température.du.thermomètre.mouillé"     
[31] "Température.du.thermomètre.mouillé"                       
[32] "Rafale.sur.les.10.dernières.minutes"                      
[33] "Rafales.sur.une.période"                                  
[34] "Periode.de.mesure.de.la.rafale"                           
[35] "Etat.du.sol"                                              
[36] "Hauteur.totale.de.la.couche.de.neige..glace..autre.au.sol"
[37] "Hauteur.de.la.neige.fraîche"                              
[38] "Periode.de.mesure.de.la.neige.fraiche"                    
[39] "Précipitations.dans.la.dernière.heure"                    
[40] "Précipitations.dans.les.3.dernières.heures"               
[41] "Précipitations.dans.les.6.dernières.heures"               
[42] "Précipitations.dans.les.12.dernières.heures"              
[43] "Précipitations.dans.les.24.dernières.heures"              
[44] "Phénomène.spécial.1"                                      
[45] "Phénomène.spécial.2"                                      
[46] "Phénomène.spécial.3"                                      
[47] "Phénomène.spécial.4"                                      
[48] "Nébulosité.couche.nuageuse.1"                             
[49] "Type.nuage.1"                                             
[50] "Hauteur.de.base.1"                                        
[51] "Nébulosité.couche.nuageuse.2"                             
[52] "Type.nuage.2"                                             
[53] "Hauteur.de.base.2"                                        
[54] "Nébulosité.couche.nuageuse.3"                             
[55] "Type.nuage.3"                                             
[56] "Hauteur.de.base.3"                                        
[57] "Nébulosité.couche.nuageuse.4"                             
[58] "Type.nuage.4"                                             
[59] "Hauteur.de.base.4"                                        
[60] "Coordonnees"                                              
[61] "Nom"                                                      
[62] "Type.de.tendance.baromé

We have various variables giving a lot of information on the weather at a given station (First variable - see https://www.data.gouv.fr/datasets/liste-des-stations-en-open-data-du-reseau-meteorologique-infoclimat-static-et-meteo-france-synop/ to understand the ID name) and a given date (Second variable)